saleor/checkout/calculations.py

```diff
@@ -258,37 +288,50 @@ def _fetch_checkout_prices_if_expired(
         if should_charge_tax:
             # Calculate taxes if charge_taxes is enabled and checkout is not exempt
             # from taxes.
-            _calculate_and_add_tax(
-                tax_calculation_strategy,
-                checkout,
-                manager,
-                checkout_info,
-                lines,
-                prices_entered_with_tax,
-                address,
-            )
+            try:
+                _calculate_and_add_tax(
+                    tax_calculation_strategy,
+                    tax_app_identifier,
+                    checkout,
+                    manager,
+                    checkout_info,
+                    lines,
+                    prices_entered_with_tax,
+                    address,
+                )
+            except EmptyTaxData:
+                if need_tax_calculation:
+                    raise ValidationError(
+                        "Configured Tax App didn't responded.",
+                        code=CheckoutErrorCode.TAX_ERROR.value,
+                    )
+                tax_app_responed = False
         else:
             # Calculate net prices without taxes.
             _get_checkout_base_prices(checkout, checkout_info, lines)
 
-    checkout.price_expiration = timezone.now() + settings.CHECKOUT_PRICES_TTL
+    checkout_update_fields = [
+        "voucher_code",
+        "total_net_amount",
+        "total_gross_amount",
+        "subtotal_net_amount",
+        "subtotal_gross_amount",
+        "shipping_price_net_amount",
+        "shipping_price_gross_amount",
+        "shipping_tax_rate",
+        "translated_discount_name",
+        "discount_amount",
+        "discount_name",
+        "currency",
+        "last_change",
+    ]
+
+    if tax_app_responed:
+        checkout.price_expiration = timezone.now() + settings.CHECKOUT_PRICES_TTL
```

Comment:
Are you sure of this? 
In case the app is down, this will cause the re-calculation of the prices for each query and mutation. It means that workers will wait for the response from the app. You already added the logic to raise an exception when taxAppID is provided, so why not leave this as it was?

Reply:
I think that this will cause to explode Saleor 😄 
If you have a query which fetches a prices, the first thing which is validating is the expiration. If your query has more than single price field to fetch, each resolver will try to call the tax app.